semantic-scope-graph/src/Control/Effect/Sketch.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Control.Effect.Sketch
+  ( Sketch (..)
+  , DeclProperties (..)
+  , RefProperties (..)
+  , declare
+  , reference
+  , Has
+  ) where
+
+import Control.Algebra
+--import Data.ScopeGraph (ScopeGraph)
+import Data.Text (Text)
+import GHC.Generics
+
+data DeclProperties = DeclProperties
+
+data RefProperties = RefProperties
+
+data Sketch address m k =
+    Declare Text DeclProperties (() -> m k)
+  | Reference Text Text RefProperties (() -> m k)
+  deriving (Generic, Generic1, HFunctor, Effect)
```

Comment:
I don’t understand the purpose of the `address` type parameter, and it seems like it makes this effect much harder to use than it would be without it (witness `AllowAmbiguousTypes`, `ScopedTypeVariables`, `TypeApplications`, and the explicit quantifiers).

Reply:
I wouldn’t bother with `DeriveAnyClass` here just for `HFunctor` & `Effect`, but that’s just personal preference.

Reply:
Yeah, I’m not sure what to do about this yet. I think we’re going to end up hardcoding `Name` values everywhere, so I vote we kill the parameter and make it concretely Name-y everywhere.

Reply:
I don’t want to step on @joshvera’s work too badly, so I’m going to address this in a follow-up patch.